lib/helper.c

```diff
@@ -647,7 +647,7 @@ uint32_t str2val32(const char *str, const struct valstr *vs)
 	int i;
 
 	for (i = 0; vs[i].str; i++) {
-		if (strncasecmp(vs[i].str, str, __maxlen(str, vs[i].str)) == 0)
+		if (strcasecmp(vs[i].str, str) == 0)
```

Comment:
The current code use val2str major on user input parameter matching to a specific number. Take 'event' code as an example, when previous application use sensor <sendorid> state <lnr|lcr|lnc|unc|ucr|unr>, so before a user can use uppercase LNR or Lnr, it will match to the predefined string and get the index. Using strcmp() will strictly use the predefined string in the code.